paddle/fluid/framework/ir/mkldnn/params_to_int8_pass.cc

```diff
@@ -0,0 +1,246 @@
+  std::string name_scope = "params_to_int8_pass";
+  GraphPatternDetector gpd;
+  patterns::Conv conv_pattern(gpd.mutable_pattern(), name_scope);
+  conv_pattern();
+
+  int params_to_int8_conv_found = 0;
+
+  auto handler = [&](const GraphPatternDetector::subgraph_t& subgraph,
+                     Graph* g) {
+    if (!IsCompat(subgraph, g)) {
+      LOG(WARNING) << "Pass in op compat failed.";
+      return;
+    }
+    VLOG(4) << "handle convolution params_to_int8_pass";
+
+    GET_IR_NODE_FROM_SUBGRAPH(conv_op, conv_op, conv_pattern);
+
+    if (!platform::HasOpINT8DataType(conv_op->Op())) {
+      return;
+    }
+
+    GET_IR_NODE_FROM_SUBGRAPH(conv_input, conv_input, conv_pattern);
+    GET_IR_NODE_FROM_SUBGRAPH(conv_filter, conv_filter, conv_pattern);
+    GET_IR_NODE_FROM_SUBGRAPH(conv_output, conv_output, conv_pattern);
+
+    auto input_names = conv_op->Op()->InputNames();
+    bool has_bias = std::find(input_names.begin(), input_names.end(), "Bias") !=
+                    input_names.end();
+    std::vector<int64_t> weights_tz = conv_filter->Var()->GetShape();
+    const int groups =
+        std::max(conv_op->Op()->GetAttrIfExists<int>("groups"), 1);
+
+    const auto& scale_weights_data =
+        conv_op->Op()->GetAttrIfExists<std::vector<float>>("Scale_weights");
+
+    bool is_multi_channel = scale_weights_data.size() > 1;
+
+    int count = 1;
+    if (is_multi_channel) {
+      count *= weights_tz[0];
+      if (groups > 1) {
+        count *= weights_tz[1];
+      }
+    }
+
+    // get scope to interact with tensors
+    auto* scope = param_scope();
+    PADDLE_ENFORCE_NOT_NULL(
+        scope, platform::errors::InvalidArgument("Scope cannot be nullptr."));
+
+    // get float weights
+    auto* weights =
+        scope->FindVar(conv_filter->Name())->GetMutable<LoDTensor>();
+
+    if (weights->dtype() != phi::DataType::FLOAT32) {
+      VLOG(4) << "Skipping convolution (id: " << conv_op->id()
+              << ") because it's a bug that it is detected again.";
+      return;
+    }
+
+    auto weights_shape = weights->dims();
+    float* weights_data = weights->data<float>();
+
+    // Create int8 weights variable
+    VarDesc int_weights(
+        patterns::PDNodeName(name_scope, "conv2d_int8_weights"));
+    int_weights.SetShape(weights_tz);
+    int_weights.SetDataType(framework::proto::VarType::Type::VarType_Type_INT8);
+    // int_weights.SetLoDLevel(conv_filter->Var()->GetLoDLevel());
+    int_weights.SetPersistable(true);
+    ir::Node* int_weights_node = g->CreateVarNode(&int_weights);
+    auto* int_weights_tensor =
+        scope->Var(int_weights_node->Name())->GetMutable<LoDTensor>();
+
+    // Initialize int8 weights
+    int_weights_tensor->Resize(weights_shape);
+    auto int_weights_data =
+        int_weights_tensor->mutable_data<int8_t>(platform::CPUPlace());
+    // Quantize weights to int8
+    auto len = weights->numel() / count;
+    if (is_multi_channel) {
```

Comment:
You can move this to one function and if `is_multi_channel` is False set count to 1 and this code will work for both cases.